clang/include/clang/AST/ExprCXX.h

```diff
@@ -5501,6 +5501,57 @@ class BuiltinBitCastExpr final
   }
 };
 
+/// Represents a C++26 reflect expression [expr.reflect]. The operand of the
+/// expression is either:
+///  - :: (global namespace),
+///  - a reflection-name,
+///  - a type-id, or
+///  - an id-expression.
+class CXXReflectExpr : public Expr {
+
+  // Source locations.
+  SourceLocation OperatorLoc;
+  SourceRange OperandRange;
+
+  CXXReflectExpr(const ASTContext &C, QualType T, QualType Ty);
+  CXXReflectExpr(const ASTContext &C, QualType T, Decl *Arg);
```

Comment:
At least avoid two `QualType` arguments with vague names.
```suggestion
  CXXReflectExpr(const ASTContext &C, QualType T, QualType Operand);
  CXXReflectExpr(const ASTContext &C, QualType T, Decl *Operand);
```